### DISCUSSION TITLE
Summary Pages

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#4495] [Development]
I'd like to create a set of summary pages for NeoMutt.

These are a set of new commands, typed at the : prompt.

| Command | Description |
| :-- | :-- |
| bind | Show a list of keybindings |
| color | Show a list of colours |
| help | Easy access to the manual |
| messages | Show a list of old mutt messages |
| scripts | Show a list of config files loaded |
| set | Show a list of variables |
| version | Show detailed version info |

**Ideas**:
- Be able to open the page in an editor
- Be able to save the info to a file
## bind -- Show a list of keybindings

**Command**: `bind [all]`
**Alternative name**: macro

Display all the keybindings and macros, grouped by NeoMutt area, e.g. Index, Pager.

With the "all" option, display everything that **can be** bound.
## color -- Show a list of colours

**Command**: `color [all]`

List all the objects that **have been** coloured.

With the "all" option, display everything that **can be** coloured.

List the colours as valid `color` commands.
Show some example text in those colours.
## help -- Easy access to the manual

**Command**: `help`

Show the text version of the manual.
This is generated by the build.

**Ideas**:
- Expande into a generic `display-file` function.
## messages -- Show a list of old mutt messages

**Command**: `messages [clear]`

Display all of the messages and errors, complete with a timestamp.

With the "clear" option, the message buffer will be emptied.
Limit the number of messages to, say, 100.

The calls to `mutt_message()` and `mutt_error()` would need to be trapped.

**Ideas**:
- Record progress bar info too, e.g. 100%, 45sec, "Opened mailbox"
- Take into account: `$read_inc` `$write_inc` `$net_inc`
## scripts -- Show a list of config files loaded

**Command**: `scripts`

Show a list of all the config files loaded, either at startup, or manually
using the source command.

Ideas:
- Red-flag any that had errors
- Add a command line option to list the files that would be loaded
- List all the non-existant files that were attempted
## set -- Show a list of variables

**Command**: `set [all]`

Display a list of all variables that have been changed.

With the option "all", display every variable.

**Ideas**:
- Highlight the variables that have been changed from the default
- Show the list as valid NeoMutt config
- Use comments to show the default values
## version -- Show detailed version info

**Command**: `version`

Show detailed version information, like `mutt -v`

**List**:
- Version
- Compilation flags
- Patches
- etc

**Ideas**:
- Add a changelog command

---

Initially, I suggest making a ":test" command which displays some sample text
in the pager.  This could be used as a template for beginners to add the other
commands.

**Comment by christopher-john-czettel** [comment]
Implemented basic scaffold for this feature-set. Even implemented :q :q! :qa and :quit commands with it. See    968b5a6 for details. Will have to figure out how to effectively and elegantly handle the whole "opening a pager for this stuff" will work best. Will continue work sometime this week.

**Comment by christopher-john-czettel** [comment]
Oh yeah: feedback and code-reviews while WIP always welcome! ;-)

**Comment by flatcap** [comment]
I've had a quick glance at the code; it looks good so far.

> Even implemented :q :q! :qa and :quit commands

I had a feeling they'd show up :-)

I have a few more ideas/notes...

| Command | Description |
| :-- | :-- |
| about | A brief description of the NeoMutt project |
| tips | Enable "tips" mode, showing brief guides |
| credits | A list of the people who created NeoMutt |

**Ideas**:
- Opening a summary page, would close an existing page / help
- Status bar shows a list of other pages, when on a summary page

**Comment by christopher-john-czettel** [comment]
just letting everyone know, that I will be working on this again. feedback and help always welcome.

**Comment by flatcap** [comment]
## hook -- show all hooks

The `:hooks` command would display a numbered list of hooks (the order is that of the config file).
After the hooks, give a list of all the folders annotated with the list of matching hook numbers.
